code/game/objects/items/barriertape.dm

```diff
@@ -0,0 +1,183 @@
+	req_access = list(ACCESS_CONSTRUCTION)
+	icon_base = "engineering"
+
+/obj/item/barrier_taperoll/attack_self(mob/user as mob)
+	if(!placing)
+		start = get_turf(src)
+		to_chat(usr, "<span class='notice'>You place the first end of the [src].</span>")
+		icon_state = "[icon_base]_stop"
+		placing = TRUE
+	else
+		placing = FALSE
+		icon_state = "[icon_base]_start"
+		end = get_turf(src)
+		if(start.y != end.y && start.x != end.x || start.z != end.z)
+			to_chat(usr, "<span class='notice'>[src] can only be laid horizontally or vertically.</span>")
+			return
+
+		var/turf/cur = start
+		var/dir
+		if (start.x == end.x)
+			var/d = end.y-start.y
+			if(d) d = d/abs(d)
+			end = get_turf(locate(end.x,end.y+d,end.z))
+			dir = "v"
+		else
+			var/d = end.x-start.x
+			if(d) d = d/abs(d)
+			end = get_turf(locate(end.x+d,end.y,end.z))
+			dir = "h"
+
+		var/can_place = TRUE
+		while (cur!=end && can_place)
+			if(cur.density || istype(cur, /turf/open/space))
+				can_place = FALSE
+			else
+				for(var/obj/O in cur)
+					if(!istype(O, /obj/structure/barrier_tape) && O.density)
+						can_place = FALSE
+						break
+			cur = get_step_towards(cur,end)
+		if (!can_place)
+			to_chat(usr, "<span class='warning'>You can't run \the [src] through that!</notice>")
+			return
+
+		cur = start
+		var/tapetest = FALSE
+		while (cur!=end)
+			for(var/obj/structure/barrier_tape/Ptest in cur)
+				if(Ptest.tape_dir == dir)
+					tapetest = TRUE
+			if(!tapetest)
+				var/obj/structure/barrier_tape/P = new tape_type(cur)
+				P.icon_state = "[P.icon_base]_[dir]"
+				P.tape_dir = dir
+			cur = get_step_towards(cur,end)
+	//is_blocked_turf(var/turf/T)
+		to_chat(usr, "<span class='notice'>You finish placing the [src].</span>")	//Git Test
+
+/obj/item/barrier_taperoll/afterattack(var/atom/A, mob/user as mob, proximity)
+	if (proximity && istype(A, /obj/machinery/door/airlock))
+		var/turf/T = get_turf(A)
+		var/obj/structure/barrier_tape/P = new tape_type(T.x,T.y,T.z)
+		P.loc = locate(T.x,T.y,T.z)
+		P.icon_state = "[icon_base]_door"
+		P.layer = 3.2
+		to_chat(user, "<span class='notice'>You finish placing the [src].</span>")
+
+/obj/structure/barrier_tape/proc/crumple()
+	if(!crumpled)
+		crumpled = TRUE
+		icon_state = "[icon_state]_c"
+		name = "crumpled [name]"
+
+/obj/structure/barrier_tape/CanAllowThrough(atom/movable/mover, turf/target)
+	. = ..()
+	if(.)
+		return
+	if(mover.pass_flags & (PASSGLASS|PASSTABLE|PASSGRILLE))
+		return 1
+	if(iscarbon(mover))
+		var/mob/living/carbon/C = mover
+		if(C.stat)	// Lets not prevent dragging unconscious/dead people.
+			return TRUE
+		if(lifted == TRUE)
+			return 1
+		if(allowed(mover))
+			return 1
+
+/obj/structure/barrier_tape/Bumped(atom/movable/AM)
+	. = ..()
+	if(iscarbon(AM))
+		add_fingerprint(AM)
+		if(lifted == FALSE)
+			to_chat(AM, "<span class='warning'>You are not supposed to go past [src]... (You can break the tape with something sharp or lift the tape with HELP intent)</span>")
```

Comment:
```suggestion
			to_chat(AM, "<span class='warning'>You are not supposed to go past [src]...</span>")
```

I think the explanation would be better on a close examination or something, not just hitting it.

Reply:
The idea was for the first month or so put it in for dumb people